ycms/cms/views/ward/ward_management_view.py

```diff
@@ -0,0 +1,94 @@
+from django.contrib import messages
+from django.shortcuts import redirect, render
+from django.utils.translation import gettext as _
+from django.views.generic import TemplateView
+
+from ...constants import job_types
+from ...forms import WardForm
+from ...models import User, Ward
+
+
+class ZentralView(TemplateView):
+    """
+    View to see all wards data and add a new one
+    """
+
+    template_name = "ward/ward_management.html"
+
+    def get(self, request, *args, **kwargs):
+        r"""
+        This function returns a list of all wards,
+        as well as a form for creating a new one
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+
+        """
+
+        all_wards = Ward.objects.all()
+        ward_form = WardForm()
+
+        return render(
+            request,
+            self.template_name,
+            {
+                "all_wards": all_wards,
+                "wards": self._get_ward_info(all_wards),
+                "ward_form": ward_form,
+                # **super().get_context_data(**kwargs),
+            },
+        )
+
+    def _get_ward_info(self, wards):
+        ward_info = {}
+
+        ward_info["all_wards_count"] = wards.count()
+
+        ward_info["all_beds_count"] = ward_info["all_wards_count"] * 28
+
+        ward_info["all_occupied_beds"] = sum(ward.occupied_beds for ward in wards)
+        ward_info["all_available_beds"] = sum(ward.available_beds for ward in wards)
+
+        ward_info["all_doctors_count"] = User.objects.filter(
+            job_type=job_types.DOCTOR
+        ).count()
+
+        ward_info["all_nurses_count"] = User.objects.filter(
+            job_type=job_types.NURSE
+        ).count()
+
+        return ward_info
+
+    def post(self, request, *args, **kwargs):
+        r"""
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :return: Redirect to list of wards
+        :rtype: ~django.http.HttpResponseRedirect
+        """
+        ward_form = WardForm(
+            data=request.POST, additional_instance_attributes={"creator": request.user}
+        )
+        if not ward_form.is_valid():
+            ward_form.add_error_messages(request)
+            return redirect("cms:protected:ward_management")
```

Comment:
Ah, this is what I meant in the previous review. Since you implemented the separate info method, we can now simply do this:
```suggestion
            return render(
                request,
                self.template_name,
                {
                    "ward_form": ward_form,
                    **self._get_ward_info(),
                    **super().get_context_data(**kwargs),
                },
            )
```
This is the same as in `get()`, but since you have instantiated the Ward form with the data the user entered, it will also be displayed with the data they entered!